validator-server/src/main/kotlin/science/numcompass/conformance/fhir/chain/TerminologyService.kt

```diff
@@ -10,6 +11,27 @@ import org.hl7.fhir.instance.model.api.IBaseResource
 import org.hl7.fhir.r4.model.ValueSet
 import science.numcompass.conformance.fhir.chain.ValidationChain.IValidator
 
+/**
+ * Try to check if a system-code pair represents a SNOMED expression that the TerminologyService
+ * will incorrectly classify as invalid. This is used for skipping validation of such terms to
+ * avoid spurious errors.
+ *
+ * Note that this function is not meant to evaluate whether a given code string is actually
+ * a *valid* SNOMED code 9or valid FHIR code in general). It will only check whether it appears
+ * to be a valid SNOMED expression code that would incorrectly fail validation. Hence, the check
+ * will actually let through e.g. valid SNOMED expressions with leading spaces. The reason is that
+ * such codes can *correctly* be flagged as invalid by the validator because FHIR codes can never
+ * have a leading space.
+ */
+internal fun isSnomedCodeThatShouldNotBeValidated(theCodeSystem: String?, theCode: String?): Boolean {
+    // Check if this is declared as a SNOMED code
+    if (theCode == null || theCodeSystem?.startsWith("http://snomed.info/sct") != true) return false
+    /* We classify any integer followed by ":", "|", or "+" (possibly with spaces)
+    as a SNOMED expression */
+    val isASnomedExpression = Regex("^\\d+\\s*[\\:\\|\\+]")
+    return isASnomedExpression.containsMatchIn(theCode)
```

Comment:
👍🏻
Would it make sense to make the RegEx a static variable and move it out the function?